examples/creatives/banner_nonlinear.js

```diff
@@ -3,36 +3,101 @@ const X_OFFSET_PERCENTAGE = .15;
 const WIDTH_PERCENTAGE = .7;
 const HEIGHT_PERCENTAGE = .15;
 
+/**
+ * A sample SIMID non-linear ad for a banner ad that shows a website when clicked on.
+ * P.S: Not all websites can be shown and they would need to allow the x-frame-options
+ *      header to sameorigin for the ad to display correctly. More information here:
+ *      https://web.dev/samesite-cookies-explained/
+ */
 class BannerNonLinear extends BaseSimidCreative {
     constructor() {
         super();
+
+        /**
+         * The desired text on the banner.
+         * @private {?string}
+         */
+        this.bannerText_ = null;
+        
+        /**
+         * The web URL to be displayed.
+         * @private {?string}
+         */
+        this.webUrl_ = null;
+
         this.addButtonClickActions_();
     }
 
     /** @override */
     onInit(eventData) {
-        super.onInit(eventData);
+        this.updateInternalOnInit(eventData);
+        this.validateAndParseAdParams_(eventData);
+        this.updateCreativeWithParams_();
         this.dynamicResize_();
     }
 
+    /**
+     * Checks validity of ad parameters and rejects with proper message if invalid.
+     * @param eventData an object that contains information details for a particular event
+     *   such as event type, unique Ids, creativeData and environmentData.
+     * @private 
+     */ 
+    validateAndParseAdParams_(eventData) {
```

Comment:
Since we're validating the adParams here as well, would we want to add some checks to make sure that (at the very least) the webUrl is included in the adParams? Something like we did with the map creative when we checked if the query was included, maybe making thne url a required param? something like [this](https://github.com/InteractiveAdvertisingBureau/SIMID/blob/master/examples/creatives/simid_map_creative.js#L75)

Reply:
yeah that sounds good